src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11782,9 +11782,20 @@ private bool IsPossibleLambdaExpression(Precedence precedence)
 
             using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
 
+            // A lambda could be starting with attributes, attempt to skip past them and check after that point.
             if (CurrentToken.Kind == SyntaxKind.OpenBracketToken)
             {
-                ParseAttributeDeclarations(inExpressionContext: true);
+                // Subtle case to deal with.  Consider:
+                //
+                // [X, () => {}       vs:
+                // [X] () => {}
```

Comment:
Should we also test cases like `[X () => {}` and `[X, Y () => {}`?

Reply:
sure.  i'll add.

Reply:
done.